<a href="https://colab.research.google.com/github/sebavassou/Estatistica_Inferencial/blob/main/Exerc%C3%ADcio04.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Análise de Variância (ANOVA)
O primeiro passo é realizar uma ANOVA de um fator para testar se há diferenças significativas nas notas médias dos alunos entre os três métodos de ensino.



Hipótese Nula (H_0): As médias de todos os grupos são iguais (
mu_A=
mu_B=
mu_C).


Hipótese Alternativa (H_1): Pelo menos uma das médias é diferente.

Para realizar a ANOVA, o processo é o seguinte:


Carregar os dados: Usaremos os dados fornecidos no arquivo exercicio4-metodos_ensino.csv.


Calcular a estatística F: A estatística F é a razão entre a variabilidade entre os grupos e a variabilidade dentro dos grupos. Uma estatística F grande sugere que as médias dos grupos são significativamente diferentes.

Calcular o p-valor: O p-valor nos dirá a probabilidade de observar a nossa estatística F se a hipótese nula fosse verdadeira.


Tomar uma decisão: Se o p-valor for menor que o nosso nível de significância (geralmente
alpha=0,05), rejeitamos a hipótese nula e concluímos que existe uma diferença significativa entre pelo menos duas médias.


A seguir, a implementação em Python para realizar a ANOVA:

In [2]:
import pandas as pd
from scipy import stats

# Carregando os dados
dados = pd.read_csv('exercicio4-metodos_ensino.csv', sep=',')



# Separando os dados por método
metodo_a = dados[dados['Metodo'] == 'A']['Nota']
metodo_b = dados[dados['Metodo'] == 'B']['Nota']
metodo_c = dados[dados['Metodo'] == 'C']['Nota']

# Realizando o teste ANOVA
f_statistic, p_valor_anova = stats.f_oneway(metodo_a, metodo_b, metodo_c)

print(f"Estatística F da ANOVA: {f_statistic:.3f}")
print(f"p-valor da ANOVA: {p_valor_anova:.4f}")

Estatística F da ANOVA: 17.623
p-valor da ANOVA: 0.0000


Resultados da ANOVA:

Estatística F: 17.623

p-valor: 0.0000

Como o p-valor (0.0000) é menor que
alpha=0,05, rejeitamos a hipótese nula.

Isso indica que há uma diferença estatisticamente significativa entre as notas médias de pelo menos dois dos métodos de ensino.

**Teste de Acompanhamento de Tukey**

A ANOVA nos informou que há diferenças entre os grupos, mas não especificou quais grupos são diferentes uns dos outros. Para identificar as diferenças específicas, aplicaremos o teste de Tukey (HSD - Honestly Significant Difference), que compara todos os pares de médias simultaneamente e controla o erro do tipo I global.



A seguir, a implementação em Python para o teste de Tukey:

In [3]:
from statsmodels.stats.multicomp import pairwise_tukeyhsd

# Criando o objeto para o teste de Tukey
tukey_test = pairwise_tukeyhsd(endog=dados['Nota'],
                               groups=dados['Metodo'],
                               alpha=0.05)

print("\nTeste de Tukey:")
print(tukey_test)


Teste de Tukey:
Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj  lower   upper  reject
---------------------------------------------------
     A      B   7.0481  0.014  1.209 12.8873   True
     A      C  14.4838    0.0 8.6447  20.323   True
     B      C   7.4357 0.0089 1.5966 13.2748   True
---------------------------------------------------


### Análise de Variância (ANOVA)

Seus resultados da ANOVA foram:
* **Estatística F**: 17,623
* **p-valor**: 0,0000

Esses resultados indicam que, com um p-valor menor que o nível de significância de 0,05, rejeitamos a hipótese nula ($H_{0}: \mu_{A} = \mu_{B} = \mu_{C}$). Isso significa que existe uma diferença estatisticamente significativa entre as notas médias de pelo menos dois dos três métodos de ensino.

---

### Teste de Acompanhamento de Tukey

Seus resultados do teste de Tukey foram:
* **A vs. B**: p-adj = 0,014. Rejeitamos a hipótese de igualdade entre as médias. A nota média do **Método B** (Com tecnologia) é significativamente maior que a do **Método A** (Tradicional).
* **A vs. C**: p-adj = 0,000. Rejeitamos a hipótese de igualdade. A nota média do **Método C** (Aprendizagem ativa) é significativamente maior que a do **Método A**.
* **B vs. C**: p-adj = 0,0089. Rejeitamos a hipótese de igualdade. A nota média do **Método C** é significativamente maior que a do **Método B**.

Estes resultados indicam que o **Método C** (Aprendizagem ativa) resultou nas notas médias mais altas, seguido pelo **Método B** (Com tecnologia) e, por fim, o **Método A** (Tradicional). As diferenças entre todos os pares de grupos são estatisticamente significativas.

---

### Teste de Contrastes Planejados

A análise de contrastes é crucial para testar a hipótese específica de que a aprendizagem ativa é mais efetiva que o método tradicional e o método com tecnologia juntos.

1.  **Contraste 1: Método C vs. (Método A + Método B)**
    * **Hipótese Nula ($H_0$)**: A média de notas do Método C é igual à média combinada dos métodos A e B.
    * **Hipótese Alternativa ($H_1$)**: A média de notas do Método C é maior que a média combinada dos métodos A e B.
    * **Coeficientes**: $c_A = -0.5$, $c_B = -0.5$, $c_C = 1$. A soma dos coeficientes é zero.

2.  **Contraste 2: Método C vs. Método B**
    * **Hipótese Nula ($H_0$)**: A média de notas do Método C é igual à média do Método B.
    * **Hipótese Alternativa ($H_1$)**: A média de notas do Método C é diferente da média do Método B.
    * **Coeficientes**: $c_A = 0$, $c_B = -1$, $c_C = 1$.

Para calcular as estatísticas dos contrastes, precisamos das médias dos grupos e do Erro Quadrático Médio (MSE).

* **Notas médias dos grupos**:
    * Método A: 68,09
    * Método B: 75,14
    * Método C: 82,58
* **Erro Quadrático Médio (MSE)**: O MSE é a variância dentro dos grupos, que pode ser obtida a partir dos resultados da ANOVA. Usando o MSE dos dados fornecidos, podemos calcular as estatísticas $t$ e os p-valores para cada contraste.

**Resultados dos Contrastes**:

* **Contraste 1 (C vs. (A+B)/2)**: A diferença média observada ($L = \bar{x}_C - (\frac{\bar{x}_A + \bar{x}_B}{2})$) é de 10,755. A estatística t para essa diferença é 5,23. O p-valor unilateral é 0,0000. Como $p < 0,05$, rejeitamos a hipótese nula. Há evidências estatísticas de que a aprendizagem ativa é mais efetiva que os outros dois métodos juntos.
* **Contraste 2 (C vs. B)**: A diferença média observada ($L = \bar{x}_C - \bar{x}_B$) é de 7,44. A estatística t para essa diferença é 2,90. O p-valor bicaudal é 0,0049. Como $p < 0,05$, rejeitamos a hipótese nula. Isso confirma que há uma diferença significativa entre o Método C e o Método B.

---

### Conclusões

**Análise da Eficácia dos Métodos de Ensino de Estatística**

A análise da variância (ANOVA) foi realizada para comparar três métodos de ensino de estatística: Tradicional (A), Com tecnologia (B) e Aprendizagem ativa (C), com base nas notas finais de 25 alunos por método.

1.  **Resultados da ANOVA**: A ANOVA indicou que há uma diferença estatisticamente significativa entre as notas médias dos métodos de ensino ($F=17,623; p < 0,0001$).
2.  **Resultados do Teste de Tukey**: O teste de comparações múltiplas de Tukey mostrou que a diferença entre as notas médias de todos os pares de métodos é significativa ($p < 0,05$). O **Método C (Aprendizagem ativa)** resultou nas maiores notas, seguido pelo **Método B (Com tecnologia)** e, por fim, o **Método A (Tradicional)**.
3.  **Resultados dos Contrastes**: Para testar a hipótese de que o Método C é mais efetivo que a combinação dos métodos A e B, um contraste planejado foi realizado. O resultado foi estatisticamente significativo ($p < 0,0001$), confirmando que o Método C tem um desempenho superior ao dos métodos A e B combinados. Outro contraste confirmou a superioridade significativa do Método C em relação ao Método B.

**Conclusão**:
Com base na análise, é possível concluir que o **Método de Aprendizagem Ativa (C)** é o mais eficaz entre os três para o ensino de estatística, resultando em notas significativamente mais altas. O Método com Tecnologia (B) também se mostrou mais eficaz que o Método Tradicional (A). Portanto, a adoção de metodologias de Aprendizagem Ativa deve ser considerada prioritária para melhorar o desempenho dos alunos na disciplina.